### PR TITLE
Header

### PR DESCRIPTION
**Pull request by MMathew93**
- Set up component Folder, created header component
- Added header to main `app.js`
- Imported logo for header
- Set `Links` for logo to route to landing page, about and search to link to landing page sections

### CODE REVIEW

**Comment by mentor-tara** [comment]
Great job :rocket: I'm a bit busy at the moment but I'll leave a review soon.

In the meantime, consider starting with the next task already to practice the professional Git workflow. [If you need a refresher, feel free to follow this section in the course again.](https://profy.dev/project/reddit-timer/while-waiting)[](ckma5leps3kko0b87rxtjjp24)

**Review by mentor-tara — CHANGES_REQUESTED** [review]
Before I approve this PR I'd like to share a few tips with you.

**Please go through the list below and review your own code first.** Imagine you're the reviewer or a person who didn't see the code before. Ask yourself if everything is clearly written and understandable. Are components and variables properly named?

Reviewing your own code with this mindset is an important practice that will help you learn a lot and write high-quality code.

To make it a bit easier for you here are the tips I promised. Some have a link :link: attached where you can find more information. I'd recommend reading the **bold tips** in any case. Feel free to check the boxes when you think your code fits.

**General:**
- [ ] Use human readable URLs [:link:](https://profy.dev/project/reddit-timer/review-tips/human-readable-links)
- [ ] Use absolute instead of relative links: `<Link to="/search" />` [:link:](https://profy.dev/project/reddit-timer/review-tips/use-absolute-links)
- [ ] No components that solely copy props to their children [:link:](https://profy.dev/project/reddit-timer/review-tips/unnecessary-components)
- [ ] Optimize svg files with SVGOMG [:link:](https://profy.dev/project/reddit-timer/review-tips/optimize-svg-with-svgomg)
- [ ] Use react-router Link instead of HTML anchor tag [:link:](https://profy.dev/project/reddit-timer/review-tips/link-instead-of-anchor-tag)

**Styles:**
- [ ] **Where to put your styles** [:link:](https://profy.dev/project/reddit-timer/review-tips/where-to-put-styles)
- [ ] **Refrain from styling children with CSS selectors** [:link:](https://profy.dev/project/reddit-timer/review-tips/css-selectors)
- [ ] **Extract shared CSS values** [:link:](https://profy.dev/project/reddit-timer/review-tips/common-css-values-in-theme)
- [ ] Be careful with pixel-perfectness when using relative font sizes like `0.9em` [:link:](https://profy.dev/project/reddit-timer/review-tips/careful-with-relative-font-size)

**Tests:**
- [ ] **Use React Testing Library for testing** [:link:](https://profy.dev/project/reddit-timer/review-tips/use-react-testing-library)
- [ ] Don't write snapshot tests [:link:](https://profy.dev/project/reddit-timer/review-tips/no-snapshot-tests)
- [ ] **Prefer integration tests over unit tests** [:link:](https://profy.dev/project/reddit-timer/review-tips/no-unit-tests)
- [ ] Get the logo by role with `getByRole` [:link:](https://profy.dev/project/reddit-timer/review-tips/get-logo-by-role)
- [ ] How to test hash links [:link:](https://profy.dev/project/reddit-timer/review-tips/testing-hash-links)

Once you're done with reviewing your own code feel free to check the [example implementation](https://profy.dev/project/reddit-timer/code-examples/header-ae7e45f0-51b2-433a-b293-3bef0ae80cce). **But remember: It's an important exercise to review your code first on your own.**

When you think this PR is ready to be merged just let me know by requesting another review. I'll approve right away.[](ckmaavvyw3m7p0b358cxn7dzq)

**Review by mentor-tara — APPROVED** [review]
Great work! :tada::tada::tada: You can merge the PR now.[](ckmabg2tk3mkv0b35j06077kr)

File: src/Views/LandingPage.js [hunk 1]
```diff
@@ -1,3 +1,47 @@
+<<<<<<< HEAD
```
Comment (jkettmann) [inline src/Views/LandingPage.js:1]: Here is a merge conflict that needs to be resolved. You can find more information about resolving conflicts [here](https://profy.dev/project/reddit-timer/merge).

I'd suggest you create a new branch `fix-header` from main (after running `git pull` there). In that branch you can resolve this conflict and the other one.

Let me know if you need more help

File: src/Views/LandingPage.js [hunk 2]
```diff
@@ -1,3 +1,47 @@
...
+  const [status, setStatus] = useState('idle');
+
+  const onSearch = async (subreddit) => {
+    setStatus('loading');
+    const url = `https://www.reddit.com/r/${subreddit}/top.json`;
+    const response = await fetch(url);
+    const { data } = await response.json();
+    setPosts(data.children);
+    setStatus('resolved');
+  };
```
Comment (jkettmann) [inline src/Views/LandingPage.js:16]: Ah interesting. This wasn't actually part of this task. The landing page only shows an image of the heatmap

Comment (jkettmann) [reply src/Views/LandingPage.js:16]: It's also best to stick to the scope of the task. The header task was only about adding the header component.

File: src/Styling/App.css [hunk 3]
```diff
@@ -2,9 +2,16 @@
     padding: 0;
     margin: 0;
     box-sizing: border-box;
+<<<<<<< HEAD
```
Comment (jkettmann) [inline src/Styling/App.css:5]: Here another conflict that needs to be resolved

File: src/components/Header.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import { Link } from 'react-router-dom';
+import Logo from '../images_and_logos/logo.svg';
+import '../Styling/Header.css';
+
+function Header() {
+  return (
+    <>
+      <header>
+        <Link to="/">
+          <img className="Logo" src={Logo} alt="Reddit Timer Logo" />
```
Comment (jkettmann) [inline src/components/Header.js:11]: Here you use global CSS classes. It's better to use CSS modules if you don't want to go for styled-components. 

It sounds more complicated than it actually is. The only thing you need to do is change the import of the css file to 

```
import styles from '../Styling/Header.css`;
```

and use the classes like this

```
<img className={styles.Logo} src={Logo} alt="Reddit Timer Logo" />
```

Now the class names are automatically scoped to the component, meaning that they get a random prefix. This way they won't conflict when you use duplicate class names inside the app.

It also makes sense to keep the style files next to the components and not in a separate `Styling` folder. I'd say that's best practice in the React world

Comment (MMathew93) [reply src/components/Header.js:11]: Oh wow I didn't even think to do that, I will keep that in mind on future projects and fix this one accordingly. I'm surprised that you wouldn't want all the style in a folder and all the pages or components in their own, though it does make sense to keep the files associated closer, I figured a folder may get too big if both the page and styles are kept in it together.

File: src/Views/LandingPage.js (hunk 2 above)
Comment (MMathew93) [reply src/Views/LandingPage.js:16]: Yeah I apologize, I think I was just following along and didn't even think to pay attention to what I was adding where and on what page. I will be correcting this today/this weekend

**Pull request merged**